### PR TITLE
[WIP - #633] Offset Functions

### PR DESCRIPTION
**Pull request by raulgrell**
For #633

I'm still looking at #1408, but I'm not sure the issue is in these functions

### CODE REVIEW

**Comment by andrewrk** [comment]
Let me know when this is ready for review

**Comment by raulgrell** [comment]
This is done, sorry about the delay!

File: test/cases/sizeof_and_typeof.zig [hunk 1]
```diff
@@ -7,28 +7,44 @@ test "sizeofAndTypeOf" {
...
-    assert(@offsetOf(P, "a") == 0);
-    assert(@offsetOf(@typeOf(p), "b") == 1);
-    assert(@offsetOf(@typeOf(p), "c") == 5);
-
-    // Non-packed struct fields can be moved/padded
-    const a: A = undefined;
-    assert(@ptrToInt(&a.a) - @ptrToInt(&a) == @offsetOf(A, "a"));
-    assert(@ptrToInt(&a.b) - @ptrToInt(&a) == @offsetOf(@typeOf(a), "b"));
-    assert(@ptrToInt(&a.c) - @ptrToInt(&a) == @offsetOf(@typeOf(a), "c"));
+    std.debug.assert(@byteOffsetOf(P, "a") == 0 and @byteOffsetOf(S, "a") == 0);
```
Comment (winksaville) [inline test/cases/sizeof_and_typeof.zig]: I thought the compiler could reorder fields for regular `struct`s, so it seems to me the bit and byte offset tests using S might be incorrect depending changes in the compiler or settings such as the target. The original tests using the difference of addresses seems like a better assertation.

**Comment by raulgrell** [comment]
You are right, the language might reorder fields and the current test is incorrect. I think ptr to int doesnt work at comptime, so either way theres a problem. The compiler isn't doing reordering yet, so I didnt worry too much for now...

**Comment by winksaville** [comment]
The previous tests were using ptrInt so I'd guess it should work on this case to.

**Pull request merged**